_posts/hardeep_nahal/2017-04-25-RBioc.md

```diff
@@ -0,0 +1,176 @@
+---
+layout: post
+title:  "Data visualization using R and Shiny"
+breadcrumb: true
+author: hardeep_nahal
+date: 2017-04-25
+categories: hardeep_nahal
+tags:
+    - Software Engineering Club
+    - R
+    - Bioconductor
+    - Shiny
+teaser:
+    info: Data visualization plays an essential role in high-throughput biology. It allows researchers to explore, understand and communicate data in a way descriptive statistics cannot compete with. 
+header:
+    version: small
+    title: Software Engineering Blog
+    image: header-logo-crop.png
+    icon: icon-blog
+---
+
+## Introduction
+
+The ability to explore large-scale datasets frequently requires the use of intuitive and interactive data visualization tools to help extract knowledge and understand complex datasets. One of the most frequently used tools in bioinformatics is [R] - a powerful statistical programming language with strong data-mining and data visualization capabilities. It offers a set of inbuilt functions and libraries to analyze and present data, such as heatmaps, boxplots, and scatter and/or line plots. It can for example, generate a heatmap to represent genes exhibiting high or low expression in certain conditions. These graphs are usually static and 2 dimensional, so sharing results usually means exporting the static format. In this blog, I will discuss two R packages called [shiny] and [d3heatmap], which can take the typical two-dimensional graph a step futher and generate an interactive web application graph built around your R analysis. This allows a user to inspect specific values by hovering the mouse over the cell, as well as zooming to a particular region of the graph by dragging a rectangle around the relevant area. In addition to generating publication-quality graphics, users can also share the graphs by simply sending a URL. 
+
+
+## What is a Shiny App?
+
+* It does not require any HTML, Javascript or CSS knowledge, but is fully extensible and customizable with HTML/Javascript/CSS/
+* Designed to integrate with existing Javascript libraries (including d3.js)
+* Provides an elegant way to make interactive R graphics and tables available via the web. It runs a web server within R, serving up interactive pages each of which can be defined by two components:
+   * ui: a user interface R script specifying the user interface connected to a computer/server running a live R session (server)
+   * server: a server R script which performs the data analysis and visualization that needs to take place on the server as the user manipulate the UI
+* A shinyApp combines the ui and server components to generate a functioning app
+
+## What is d3heatmap?
+* Generates interactive heatmaps 
+* Allows a user to click on rows/columns, or draw rectangles to zoom in
+
+
```

Comment:
It might be better to move the first phrase of "Creating an interactive web-based heatmap" before "Getting Started", this way you explain what you are trying to achieve before going into the technical details of the execution.